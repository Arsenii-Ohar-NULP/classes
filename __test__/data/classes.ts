import Class from 'components/classes/Class';

export const sampleFiveClasses: Class[] = [
    {
        id: 1,
        title: 'Class 1',
        description: 'This is class 1',
        teacher_id: 1,
        students_number: 1
    },
    {
        id: 2,
        title: 'Class 2',
        description: 'This is class 2',
        teacher_id: 2,
        students_number: 1
    },
    {
        id: 3,
        title: 'Class 3',
        description: 'This is class 3',
        teacher_id: 3,
        students_number: 1

    },
    {
        id: 4,
        title: 'Class 4',
        description: 'This is class 4',
        teacher_id: 4,
        students_number: 1
    },
    {
        id: 5,
        title: 'Class 5',
        description: 'This is class 5',
        teacher_id: 5,
        students_number: 1
    }
]