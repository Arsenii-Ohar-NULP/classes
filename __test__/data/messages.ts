import Message from 'pages/class/Message';

const sampleMessages : Message[] = [
    {
        id: 1,
        content: 'First',
        date: 'Yesterday',
        user: 1,
        cls: 1,
        fullname: 'Arsenii Ohar',
        username: 'arseniyohar'
    },
    {
        id: 2,
        content: 'First',
        date: 'Yesterday',
        user: 1,
        cls: 1,
        fullname: 'Arsenii Ohar',
        username: 'arseniyohar'
    },
    {
        id: 3,
        content: 'First',
        date: 'Yesterday',
        user: 1,
        cls: 1,
        fullname: 'Arsenii Ohar',
        username: 'arseniyohar'
    }

]