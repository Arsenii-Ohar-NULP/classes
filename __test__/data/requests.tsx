import { JoinRequest } from 'pages/class/JoinRequest'

export const getRequestsWithClassId = (classId: number, number: number) => {
    const requests: JoinRequest[] = []
    for (let i = 0; i < number; i++){
        requests.push(
            {
                userId: i + 1,
                classId: classId
            }
        )
    }

    return requests;
}