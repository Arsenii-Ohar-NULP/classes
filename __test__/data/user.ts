import User, { Role } from 'components/account/User';

export const sampleUser: User = {
    id: 1,
    username: 'seniorohar',
    firstName: 'Arsenii',
    lastName: 'Ohar',
    email: 'arsen.ogar@gmail.com',
    phone: '+380987669293',
    role: Role.Teacher   
}

export const sampleFiveUsers: User[] = [
    {
        id: 1,
        username: 'seniorohar',
        firstName: 'Arsenii',
        lastName: 'Ohar',
        email: 'arsen.ogar@gmail.com',
        phone: '+380987669293',
        role: Role.Student
    },
    {
        id: 2,
    username: 'vitalikpikh',
    firstName: 'Vitalik',
    lastName: 'Pikhotskiy',
    email: 'vitaliipik@gmail.com',
    phone: '+380967629253',
    role: Role.Student
    },
    {
        id: 3,
    username: 'svyatoslavshainoha',
    firstName: 'Svyatoslav',
    lastName: 'Shainoha',
    email: 'vitaliipik@gmail.com',
    phone: '+380967629253',
    role: Role.Student
    },
      {
        id: 4,
    username: 'olehrysin',
    firstName: 'Oleh',
    lastName: 'Rysin',
    email: 'olehrysin@gmail.com',
    phone: '+380967323273',
    role: Role.Student
    },
    {
        id: 5,
    username: 'johnadams',
    firstName: 'John',
    lastName: 'Adams',
    email: 'johnadams@gmail.com',
    phone: '+380967723273',
    role: Role.Student
    }
]