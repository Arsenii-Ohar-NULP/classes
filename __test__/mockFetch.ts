export const mockFetch = () => {
    global.fetch = jest.fn();
}