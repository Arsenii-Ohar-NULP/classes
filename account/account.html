<!DOCTYPE html>
<!--[if lt IE 7]>      <html class="no-js lt-ie9 lt-ie8 lt-ie7"> <![endif]-->
<!--[if IE 7]>         <html class="no-js lt-ie9 lt-ie8"> <![endif]-->
<!--[if IE 8]>         <html class="no-js lt-ie9"> <![endif]-->
<!--[if gt IE 8]>      <html class="no-js"> <!--<![endif]-->
<html>
  <head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <title>Account</title>
    <meta name="description" content="" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <link rel="stylesheet" href="account.css" />
  </head>
  <body>
    <!--[if lt IE 7]>
      <p class="browsehappy">
        You are using an <strong>outdated</strong> browser. Please
        <a href="#">upgrade your browser</a> to improve your experience.
      </p>
    <![endif]-->
    <nav class="navbar navbar-expand-lg bg-body-tertiary" data-bs-theme="dark">
      <div class="container-fluid">
        <a class="navbar-brand" href="#">Classes</a>
        <button
          class="navbar-toggler"
          type="button"
          data-bs-toggle="collapse"
          data-bs-target="#navbarNavDropdown"
          aria-controls="navbarNavDropdown"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span class="navbar-toggler-icon"></span>
        </button>
        <div
          class="collapse navbar-collapse justify-content-between"
          id="navbarNavDropdown"
        >
          <ul class="navbar-nav">
            <li class="nav-item">
              <a class="nav-link" aria-current="page" href="#">Home</a>
            </li>
          </ul>
          <ul class="navbar-nav">
            <li class="nav-item">
              <a class="nav-link" href="#">
                <img
                  src="images/how-to-watch-john-wick (2).jpg"
                  height="48"
                  width="48"
                  class="rounded-circle"
                />
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
    <div class="container">
      <p class="fs-3 p-3">Account</p>

      <form class="">
        <div class="container card bg-light h-100 rounded-5 m-2">
          <div class="container m-1 text-center">
            <img
              src="./images/how-to-watch-john-wick (2).jpg"
              class="rounded-circle p-3"
            />
            <div class="input-group m-1 my-3">
              <label class="input-group-text" for="inputGroupFile01"
                >Upload</label
              >
              <input
                type="file"
                class="form-control"
                id="inputGroupFile01"
                accept=".gif,.jpg,.jpeg,.png"
              />
            </div>
          </div>
          <div class="p-1">
            <label for="nicknameControl" class="form-label">Nickname</label>
            <input
              type="tel"
              class="form-control"
              id="nicknameControl"
              placeholder="johnsmith"
              value="johnwick"
            />
          </div>
          <div class="p-1">
            <div class="row">
              <div class="col-lg">
                <label for="firstNameForm" class="form-label">First name</label>
                <input
                  type="text"
                  class="form-control m-1"
                  id="firstNameForm"
                  placeholder="John"
                  value="John"
                />
              </div>
              <div class="col-lg">
                <label for="lastNameForm" class="form-label">Last name</label>
                <input
                  type="text"
                  class="form-control m-1"
                  id="lastNameForm"
                  placeholder="Smith"
                  value="Wick"
                />
              </div>
            </div>
          </div>
          <div class="p-1">
            <label for="emailFormControl" class="form-label"
              >Email address</label
            >
            <input
              type="email"
              class="form-control"
              id="emailFormControl"
              placeholder="name@email.com"
              value="johnwick@gmail.com"
            />
          </div>
          <div class="p-1">
            <label for="phoneNumberControl" class="form-label"
              >Phone number</label
            >
            <input
              type="tel"
              class="form-control"
              id="phoneNumberControl"
              placeholder="+38098000000"
              value="+380987669293"
            />
          </div>
          <div class="container mx-auto my-3 text-center">
            <button type="button" class="btn btn-primary">Save</button>
          </div>
        </div>
      </form>
    </div>
    <script src="" async defer></script>
  </body>
</html>
