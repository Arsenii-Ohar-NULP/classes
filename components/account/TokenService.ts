export const accessTokenKey = 'accessToken';

export function getAccessToken(): string {
    const localToken = localStorage.getItem(accessTokenKey);
    const sessionToken = sessionStorage.getItem(accessTokenKey);

    return localToken ? localToken : sessionToken;
}