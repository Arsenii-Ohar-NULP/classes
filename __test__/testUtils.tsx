import React, { PropsWithChildren } from 'react';
import { render, renderHook } from '@testing-library/react';
import type { RenderOptions } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import type { PreloadedState } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import type { RootState, AppStore } from 'pages/redux/store';
// As a basic setup, import your same slice reducers
import { authInitialState, authSlice } from 'pages/redux/auth';
import { classSlice, classesInitialState } from 'pages/redux/classes';

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = { auth: authInitialState, classes: classesInitialState },
    // Automatically create a store instance if no store was passed in
    store = configureStore({
      reducer: { auth: authSlice.reducer, classes: classSlice.reducer },
      preloadedState,
    }),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<object>): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export function renderHookWithProviders(
  ui,
  {
    preloadedState = { auth: authInitialState, classes: classesInitialState },
    // Automatically create a store instance if no store was passed in
    store = configureStore({
      reducer: { auth: authSlice.reducer, classes: classSlice.reducer },
      preloadedState,
    }),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<object>): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...renderHook(() => ui(), { wrapper: Wrapper, ...renderOptions }) };
}
