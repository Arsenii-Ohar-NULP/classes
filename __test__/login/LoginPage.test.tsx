import React from 'react';
import {fireEvent, screen, waitFor} from '@testing-library/react';

import LoginPage from 'app/auth/login/page';
import {renderWithProviders} from '../testUtils';
import {LocalStorageMock} from '__test__/LocalStorageMock';
import {saveToken} from 'components/login/AuthService';
import {sampleUser} from '__test__/data/user';
import {useRouter} from 'next/navigation';
import {getAccessToken} from "components/account/TokenService";
import {server} from "../api/server";
import {CLASSES_API_URL} from "components/redux/utils";
import {rest} from "msw";

jest.mock('components/account/TokenService')
jest.mock('components/login/AuthService', () => ({
    useLogout: () => jest.fn(),
    saveToken: jest.fn(),

}))
const mockNavigate = jest.fn();
jest.mock('next/navigation', () => ({
    useRouter: () => ({
        replace: mockNavigate,
        push: mockNavigate
    })
}));

Object.defineProperty(window, 'localStorage', {
    value: new LocalStorageMock(),
});


describe('Login Page', () => {
    beforeEach(() => {
        window.localStorage.clear();
        jest.clearAllMocks();
    });

    const fillUsername = (username: string) => {
        const usernameInput = screen.getByTestId('Username');
        fireEvent.change(usernameInput, {target: {value: username}});
    };

    const fillPassword = (password: string) => {
        const passwordInput = screen.getByTestId('Password');
        fireEvent.change(passwordInput, {target: {value: password}});
    };

    const fillUpForm = () => {
        const sampleUsername = 'senirohar';
        const samplePassword = 'qwerty1234';

        fillUsername(sampleUsername);
        fillPassword(samplePassword);
    };

    const mockValidUser = (token: string) => {
        const user = sampleUser;
        server.use(rest.post(`${CLASSES_API_URL}/user/login`, (req, res, ctx) => {
            return res(ctx.status(200), ctx.json({access_token: token}))
        }), rest.get(`${CLASSES_API_URL}/user`, (req, res, ctx) => {
            return res(ctx.status(200), ctx.json(user))
        }));

        return user;
    }

    it('should match a snapshot', async () => {
        const page = renderWithProviders(<LoginPage/>);
        expect(page.container).toMatchSnapshot();
    });

    it('when token exists, should push to /main/classes', async () => {
        const token = 'Q2312312321311225';
        jest.mocked(getAccessToken).mockImplementationOnce(() => token);
        server.use(rest.get(`${CLASSES_API_URL}/user`, (req, res, ctx) => {
            return res(ctx.status(200), ctx.json(sampleUser));
        }))
        renderWithProviders(<LoginPage/>);
        await waitFor(() => {
            expect(jest.mocked(useRouter().push)).toBeCalledWith('/main/classes');
            expect(jest.mocked(getAccessToken)).toBeCalled();
        });
    });

    it('when signup button is clicked, should push to sign up page', async () => {
        renderWithProviders(<LoginPage/>);

        const signUpButton = screen.getByText('Sign up');
        fireEvent.click(signUpButton);

        await waitFor(() =>
            expect(jest.mocked(useRouter().push)).toBeCalledWith('/auth/signUp')
        );
    });

    it('when validation error occurs, shows fail message', async () => {
        const mockedLogin = jest.fn();
        const sampleUsername = 'hell';
        mockedLogin.mockReturnValueOnce('ABCD123');
        jest.mock('components/login/AuthService', () => ({
            login: mockedLogin,
        }));

        renderWithProviders(<LoginPage/>);
        fillUsername(sampleUsername);
        const login = screen.getByText('Log in');
        fireEvent.click(login);

        await waitFor(() => expect(mockedLogin).not.toBeCalled());
        const error = screen.getByTestId('Username-Error');
        expect(error).toHaveTextContent('');
    });

    it('when backend error, show server error', async () => {
        const errorMessage = "Something stupid happened";
        server.use(rest.post(`${CLASSES_API_URL}/user/login`, (req, res, ctx) => {
            return res(ctx.status(401), ctx.json({msg: errorMessage}));
        }))

        renderWithProviders(<LoginPage/>);
        fillUpForm();
        const loginButton = screen.getByText('Log in');
        fireEvent.click(loginButton);
        await waitFor(() => {
            const error = screen.getByTestId('error');
            expect(error).toBeInTheDocument();
            screen.getByText(errorMessage);
        });
    });

    it('when username set and sign up is clicked, should push to /signUp/?username=', async () => {
        renderWithProviders(<LoginPage/>);
        const usernameInput = screen.getByTestId('Username');
        const username = 'seniorohar';
        fireEvent.change(usernameInput, {target: {value: username}});
        const signUpButton = screen.getByText('Sign up');
        fireEvent.click(signUpButton);

        await waitFor(() =>
            expect(jest.mocked(useRouter().push)).toBeCalledWith(
                `/auth/signUp/?username=${username}`
            )
        );
    });

    it('when rememberMe is unchecked and clicked log in with valid data, should call save token with remember me to false', async () => {
        const tickCheckbox = () => {
            const checkbox = screen.getByTestId('remember-me');
            fireEvent.click(checkbox);
        };
        const token = 'ABCD1234';
        const rememberMe = false;

        mockValidUser(token);

        renderWithProviders(<LoginPage/>);
        fillUpForm();
        tickCheckbox();
        const loginButton = screen.getByText('Log in');
        fireEvent.click(loginButton);
        await waitFor(() => {
            expect(jest.mocked(saveToken)).toBeCalledWith(token, rememberMe);
        });
    });

    it('when tries to log in on valid data, should fetch the access token and no error should occur', async () => {
        const token = 'ABCD1234';
        mockValidUser(token);
        renderWithProviders(<LoginPage/>);
        const usernameInput = screen.getByTestId('Username');
        fireEvent.change(usernameInput, {target: {value: 'seniorohar'}});
        const passwordInput = screen.getByTestId('Password');
        fireEvent.change(passwordInput, {target: {value: 'gjasgjasjg1234'}});
        const loginButton = screen.getByText('Log in');
        fireEvent.click(loginButton);

        await waitFor(() => {
            expect(screen.getByTestId('Username-Error')).toBeEmptyDOMElement();
            expect(screen.getByTestId('Password-Error')).toBeEmptyDOMElement();
            expect(useRouter().replace).toHaveBeenCalledWith("/main/classes");
        });
    });
});
