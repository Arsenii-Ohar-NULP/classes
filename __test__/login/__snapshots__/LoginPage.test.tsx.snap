// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page should match a snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="d-flex w-100"
      >
        <div
          class="container-lg justify-content-center align-items-center text-center"
        >
          <form
            class="main needs-validation"
            id="form"
            name="form"
          >
            <div
              class="p-2"
            >
              <div
                class="d-flex justify-content-center align-items-center p-2"
              >
                <div
                  class="d-flex justify-content-center align-items-center gap-1 bg-dark p-2 rounded"
                >
                  <img
                    alt="Classes Logo"
                    class="border rounded-3 p-1 m-0 fill-light classes-pic"
                    data-nimg="1"
                    decoding="async"
                    height="32"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
                    style="color: transparent;"
                    width="32"
                  />
                  <h3
                    class="m-0 align-middle text-primary"
                  >
                    Classes
                  </h3>
                </div>
              </div>
              <hr
                class="w-100"
              />
            </div>
            <p
              class="fs-5 position-relative"
            >
              <b>
                Hi, please log in 
              </b>
              <img
                alt="Smiling Face"
                data-nimg="1"
                decoding="async"
                height="28"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
                style="color: transparent;"
                width="28"
              />
            </p>
            <div
              class="my-2 has-validation"
            >
              <input
                class="form-control-lg rounded-4 p-1 px-2"
                data-testid="Username"
                id="Username"
                name="Username"
                placeholder="Enter a username"
                type="text"
              />
              <div
                class="invalid"
                data-testid="Username-Error"
              />
            </div>
            <div
              class="my-2 has-validation"
            >
              <input
                class="form-control-lg rounded-4 p-1 px-2"
                data-testid="Password"
                id="Password"
                name="Password"
                placeholder="Enter a password"
                type="password"
              />
              <div
                class="invalid"
                data-testid="Password-Error"
              />
            </div>
            <div
              class="form-check d-inline-block justify-content-center align-items-center"
            >
              <input
                checked=""
                class="form-check-input"
                data-testid="remember-me"
                formnovalidate=""
                id="remember-me"
                name="remember-me"
                type="checkbox"
              />
              <label
                class="form-check-label"
                for="remember-me"
              >
                Remember me
              </label>
            </div>
            <p
              class="visually-hidden"
              data-testid="error"
            />
            <div
              class="d-flex my-2 gap-2 h-50 justify-content-center"
            >
              <button
                class="btn btn-primary rounded-pill px-3"
                id="loginButton"
                type="button"
              >
                Log in
              </button>
              <div
                class="vr m-1"
              />
              <button
                class="btn btn-secondary rounded-pill px-3"
                type="button"
              >
                Sign up
              </button>
            </div>
          </form>
          <footer />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="d-flex w-100"
    >
      <div
        class="container-lg justify-content-center align-items-center text-center"
      >
        <form
          class="main needs-validation"
          id="form"
          name="form"
        >
          <div
            class="p-2"
          >
            <div
              class="d-flex justify-content-center align-items-center p-2"
            >
              <div
                class="d-flex justify-content-center align-items-center gap-1 bg-dark p-2 rounded"
              >
                <img
                  alt="Classes Logo"
                  class="border rounded-3 p-1 m-0 fill-light classes-pic"
                  data-nimg="1"
                  decoding="async"
                  height="32"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
                  style="color: transparent;"
                  width="32"
                />
                <h3
                  class="m-0 align-middle text-primary"
                >
                  Classes
                </h3>
              </div>
            </div>
            <hr
              class="w-100"
            />
          </div>
          <p
            class="fs-5 position-relative"
          >
            <b>
              Hi, please log in 
            </b>
            <img
              alt="Smiling Face"
              data-nimg="1"
              decoding="async"
              height="28"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
              style="color: transparent;"
              width="28"
            />
          </p>
          <div
            class="my-2 has-validation"
          >
            <input
              class="form-control-lg rounded-4 p-1 px-2"
              data-testid="Username"
              id="Username"
              name="Username"
              placeholder="Enter a username"
              type="text"
            />
            <div
              class="invalid"
              data-testid="Username-Error"
            />
          </div>
          <div
            class="my-2 has-validation"
          >
            <input
              class="form-control-lg rounded-4 p-1 px-2"
              data-testid="Password"
              id="Password"
              name="Password"
              placeholder="Enter a password"
              type="password"
            />
            <div
              class="invalid"
              data-testid="Password-Error"
            />
          </div>
          <div
            class="form-check d-inline-block justify-content-center align-items-center"
          >
            <input
              checked=""
              class="form-check-input"
              data-testid="remember-me"
              formnovalidate=""
              id="remember-me"
              name="remember-me"
              type="checkbox"
            />
            <label
              class="form-check-label"
              for="remember-me"
            >
              Remember me
            </label>
          </div>
          <p
            class="visually-hidden"
            data-testid="error"
          />
          <div
            class="d-flex my-2 gap-2 h-50 justify-content-center"
          >
            <button
              class="btn btn-primary rounded-pill px-3"
              id="loginButton"
              type="button"
            >
              Log in
            </button>
            <div
              class="vr m-1"
            />
            <button
              class="btn btn-secondary rounded-pill px-3"
              type="button"
            >
              Sign up
            </button>
          </div>
        </form>
        <footer />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`this is a Login Page test matches a snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="d-flex w-100"
      >
        <div
          class="container-lg justify-content-center align-items-center text-center"
        >
          <form
            class="main needs-validation"
            id="form"
            name="form"
          >
            <div
              class="p-2"
            >
              <div
                class="d-flex justify-content-center align-items-center p-2"
              >
                <div
                  class="d-flex justify-content-center align-items-center gap-1 bg-dark p-2 rounded"
                >
                  <img
                    alt="Classes Logo"
                    class="border rounded-3 p-1 m-0 fill-light classes-pic"
                    data-nimg="1"
                    decoding="async"
                    height="32"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
                    style="color: transparent;"
                    width="32"
                  />
                  <h3
                    class="m-0 align-middle text-primary"
                  >
                    Classes
                  </h3>
                </div>
              </div>
              <hr
                class="w-100"
              />
            </div>
            <p
              class="fs-5 position-relative"
            >
              <b>
                Hi, please log in 
              </b>
              <img
                alt="Smiling Face"
                data-nimg="1"
                decoding="async"
                height="28"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
                style="color: transparent;"
                width="28"
              />
            </p>
            <div
              class="my-2 has-validation"
            >
              <input
                class="form-control-lg rounded-4 p-1 px-2"
                data-testid="Username"
                id="Username"
                name="Username"
                placeholder="Enter a username"
                type="text"
              />
              <div
                class="invalid"
                data-testid="Username-Error"
              />
            </div>
            <div
              class="my-2 has-validation"
            >
              <input
                class="form-control-lg rounded-4 p-1 px-2"
                data-testid="Password"
                id="Password"
                name="Password"
                placeholder="Enter a password"
                type="password"
              />
              <div
                class="invalid"
                data-testid="Password-Error"
              />
            </div>
            <div
              class="form-check d-inline-block justify-content-center align-items-center"
            >
              <input
                checked=""
                class="form-check-input"
                data-testid="remember-me"
                formnovalidate=""
                id="remember-me"
                name="remember-me"
                type="checkbox"
              />
              <label
                class="form-check-label"
                for="remember-me"
              >
                Remember me
              </label>
            </div>
            <p
              class="visually-hidden"
              data-testid="error"
            />
            <div
              class="d-flex my-2 gap-2 h-50 justify-content-center"
            >
              <button
                class="btn btn-primary rounded-pill px-3"
                id="loginButton"
                type="button"
              >
                Log in
              </button>
              <div
                class="vr m-1"
              />
              <button
                class="btn btn-secondary rounded-pill px-3"
                type="button"
              >
                Sign up
              </button>
            </div>
          </form>
          <footer />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="d-flex w-100"
    >
      <div
        class="container-lg justify-content-center align-items-center text-center"
      >
        <form
          class="main needs-validation"
          id="form"
          name="form"
        >
          <div
            class="p-2"
          >
            <div
              class="d-flex justify-content-center align-items-center p-2"
            >
              <div
                class="d-flex justify-content-center align-items-center gap-1 bg-dark p-2 rounded"
              >
                <img
                  alt="Classes Logo"
                  class="border rounded-3 p-1 m-0 fill-light classes-pic"
                  data-nimg="1"
                  decoding="async"
                  height="32"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
                  style="color: transparent;"
                  width="32"
                />
                <h3
                  class="m-0 align-middle text-primary"
                >
                  Classes
                </h3>
              </div>
            </div>
            <hr
              class="w-100"
            />
          </div>
          <p
            class="fs-5 position-relative"
          >
            <b>
              Hi, please log in 
            </b>
            <img
              alt="Smiling Face"
              data-nimg="1"
              decoding="async"
              height="28"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
              style="color: transparent;"
              width="28"
            />
          </p>
          <div
            class="my-2 has-validation"
          >
            <input
              class="form-control-lg rounded-4 p-1 px-2"
              data-testid="Username"
              id="Username"
              name="Username"
              placeholder="Enter a username"
              type="text"
            />
            <div
              class="invalid"
              data-testid="Username-Error"
            />
          </div>
          <div
            class="my-2 has-validation"
          >
            <input
              class="form-control-lg rounded-4 p-1 px-2"
              data-testid="Password"
              id="Password"
              name="Password"
              placeholder="Enter a password"
              type="password"
            />
            <div
              class="invalid"
              data-testid="Password-Error"
            />
          </div>
          <div
            class="form-check d-inline-block justify-content-center align-items-center"
          >
            <input
              checked=""
              class="form-check-input"
              data-testid="remember-me"
              formnovalidate=""
              id="remember-me"
              name="remember-me"
              type="checkbox"
            />
            <label
              class="form-check-label"
              for="remember-me"
            >
              Remember me
            </label>
          </div>
          <p
            class="visually-hidden"
            data-testid="error"
          />
          <div
            class="d-flex my-2 gap-2 h-50 justify-content-center"
          >
            <button
              class="btn btn-primary rounded-pill px-3"
              id="loginButton"
              type="button"
            >
              Log in
            </button>
            <div
              class="vr m-1"
            />
            <button
              class="btn btn-secondary rounded-pill px-3"
              type="button"
            >
              Sign up
            </button>
          </div>
        </form>
        <footer />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
